client/elements/lookups/sc-lzh2en.html

```diff
@@ -1,20 +1,126 @@
+<link rel="import" href="/bower_components/iron-ajax/iron-ajax.html">
+<link rel="import" href="/bower_components/iron-meta/iron-meta.html">
 
 <dom-module id="sc-chinese-lookup">
-  <template>
+    <template>
+        <iron-ajax
+        id="ajax"
+        handle-as="json"
+        loading="{{loadingDict}}"
+        last-response="{{dictData}}"></iron-ajax>
 
-  </template>
+        <iron-ajax
+        id="fallback_ajax"
+        handle-as="json"
+        loading="{{loadingFallbackDictData}}"
+        last-response="{{fallbackDictData}}"></iron-ajax>
 
-  <script>
-      class SCChineseLookup extends Polymer.Element {
-          static get is() {
-              return 'sc-chinese-lookup';
-          }
+        <iron-meta id="meta"></iron-meta>
 
-          static get properties() {}
-      }
+    </template>
 
-      customElements.define(SCChineseLookup.is, SCChineseLookup);
-  </script>
-</dom-module>
+    <script>
+        class SCChineseLookup extends ReduxMixin(Polymer.Element) {
+            static get is() {
+                return 'sc-chinese-lookup';
+            }
 
+            static get properties() {
+                return {
+                    loadedLanguage: {
+                        type: String
+                    },
+                    toLang: {
+                        type: String,
+                        statePath: 'textOptions.chineseLookupTargetLanguage',
+                        observer: '_targetLanguageChanged'
+                    },
+                    dictData: {
+                        type: Object,
+                    },
+                    loadingDict: {
+                        type: Boolean,
+                        value: true
+                    },
+                    loadingFallbackDictData: {
+                        type: Boolean,
+                        value: true
+                    },
+                    fallbackDictData: {
+                        type: Object
+                    }
+                }
+            }
+
+            getNewDict() {
+                this.$.ajax.url = this._computeUrl();
+                this.loadedLanguage = this.toLang;
+                if(this.toLang === 'en'){
+                    this.$.fallback_ajax.url = this._computeUrl(true);
+                    this.$.fallback_ajax.generateRequest();
+                }
+                return this.$.ajax.generateRequest();
+            }
+
+            lookupWord(graphs) {
+                let definition = '';
+                let length = 1;
+                for (let i = graphs.length; i > 0; i--) {
+                    let snip = graphs.slice(0, i);
+                    definition += this._lookupWord(snip);
+                    if(definition && i > length){
+                        length = i;
+                    }
+                }
+                return {html: definition, length: length};
+            }
+
+            _lookupWord(graph){
+                graph = graph.replace(/\u2060/, '');
```

Comment:
what's this symbol? Maybe extract it to a variable?

Reply:
No idea what it is